### PR TITLE
Newly built mqtt to output

### PR DESCRIPTION
**Pull request by davidgs**
mqtt.to() output from flux

### Done checklist
- [ ] docs/SPEC.md updated
- [ x] Test cases written

### CODE REVIEW

File: stdlib/mqtt/to.go [hunk 1]
```diff
@@ -0,0 +1,649 @@
...
+	opts := MQTT.NewClientOptions().AddBroker(t.spec.Spec.Broker)
+	if len(t.spec.Spec.ClientID) > 0 {
+		opts.SetClientID(t.spec.Spec.ClientID)
+	} else {
+		opts.SetClientID("flux-mqtt")
+	}
+	if t.spec.Spec.Timeout > 0 {
+		opts.SetConnectTimeout(t.spec.Spec.Timeout)
+	}
+	if len(t.spec.Spec.Username) > 0 {
```
Comment (aanthony1243) [inline stdlib/mqtt/to.go]: if these are strings, it might be clearer to say `if t.spec.Spec.Username != ""`

File: stdlib/mqtt/to.go [hunk 2]
```diff
@@ -0,0 +1,649 @@
...
+			l := er.Len()
+			for i := 0; i < l; i++ {
+				m.truncateTagsAndFields()
+				for j, col := range er.Cols() {
+					switch {
+					case col.Label == timeColLabel:
+						m.t = values.Time(er.Times(j).Value(i)).Time()
+					case measurementNameCol != "" && measurementNameCol == col.Label:
+						if col.Type != flux.TString {
+							return errors.New("invalid type for measurement column")
```
Comment (aanthony1243) [inline stdlib/mqtt/to.go]: these types of errors should be checked outside of tbl.Do  using `tbl.Cols()`.  Doing this check over and over for every row will slow you down quite a bit.

File: stdlib/mqtt/to.go [hunk 3]
```diff
@@ -0,0 +1,649 @@
...
+	if err := wg.Wait(); err != nil {
+		client.Disconnect(250)
+		return err
+	}
+	client.Disconnect(250)
+	return nil
+
+}
+// format the message as json
+func (t *toMqttMetric) formatJSON(message string) strings.Builder {
```
Comment (aanthony1243) [inline stdlib/mqtt/to.go]: this is sort of an antipattern in Go.  You can use struct annotations to marshal a simple struct as JSON.   so you have something like   `type metric struct{...}` that can be encoded properly as JSON, and then you can have a `func (m metric) toMessage() string {...}`   to encode the message as a simple string.

File: stdlib/mqtt/to.go (hunk 1 above)
Comment (davidgs) [reply stdlib/mqtt/to.go]: Done!

File: stdlib/mqtt/to.go (hunk 2 above)
Comment (davidgs) [reply stdlib/mqtt/to.go]: Sounds like we should leave this as-is.

**Comment by davidgs** [comment]
Pulling out JSON support, re-working the tests, etc. Will re-submit as a new PR.

**Pull request closed without merging**